### ISSUE TITLE
scoll down not working properly / jerky

### ISSUE THREAD

**Issue by TurboGit** [darktable-org/darktable#11113]
**Describe the bug/issue**

The scroll down is not progressive and seems to have some blocking / going back. One need to move fast and longer to move down in a list.

**To Reproduce**
1. Go to 'collect' module
2. Scroll slowly down to move down in the list
3. See the jerky move or no move at all
4. Scroll fast, the move will happen even if a bit jerky
5. Try moving up, no issue very smooth and responsive

This is not related to collect module as the same can be reproduced with the styles module.

It has been some time since I have this behavior. I first thought that it was due to some Gtk issue on my side (I'm using GNU/Debian sid). But it has persisted for long enough for not being the cause and I don't have such issue with other software.

I'm wondering if this is not due to new Input-NG system.

**Expected behavior**

Smooth scrolling when moving down.

**Platform**

* darktable version : master
* OS                : GNU/Linux
* Linux - Distro    : GNU/Debian - sid

**Comment by dterrahe** [comment]
> I suspect the Input-NG to be in play here.

Why? Are you scrolling _while_ holding a shortcut key?

Would you please clarify for me; are you talking about scrolling in _listviews_ only? Is this with lists with many items only or with few as well? In _resizable_ listviews (ctrl-scroll) only or all of them? I'm asking because I'm not obviously seeing the issue so I may need some help reproducing.

(The candidate I was thinking of is 8c31abfd716b8dc6a0c7a07be6d6f195298d679c, but it may be completely unrelated as well).

If somebody else recognises the issue and is able to do a bisect, that would be very helpful!

**Comment by TurboGit** [comment]
> Why? Are you scrolling _while_ holding a shortcut key?

No. But just tested and also reproduce with 3.6. So that's not Input-NG.

> Would you please clarify for me; are you talking about scrolling in listviews only?

Yes.

> Is this with lists with many items only or with few as well?

Yes, at least to have the scrollbar activated.

> In resizable listviews (ctrl-scroll) only or all of them?

Good question, but I'm not sure we have a non resizable listview in dt to test at this stage. Do we?

The candidate commit is indeed a possibility. Will test later.

**Comment by TurboGit** [comment]
Ok, https://github.com/darktable-org/darktable/commit/8c31abfd716b8dc6a0c7a07be6d6f195298d679c is not the culprit.

**Comment by TurboGit** [comment]
Just in case it helps, when more you scroll slowly and easier it is possible to reproduce. When I scroll very slowly I see the listview "jerk" and actually not moving of position as if an action made it back to it's original location.

**Comment by TurboGit** [comment]
> When you resize the textview, does it fit a whole number of lines or do you see partial/cut off lines?

It always fit a whole number of line and the padding at the bottom is very uniform whatever I resize the textview. So this part looks good.

**Comment by TurboGit** [comment]
After

`    value -= value % increment;`

Adding the following code fix the issue:

```
    if(value == before)
    {
      if(event->delta_y > 0)
        value += increment;
      else
        value -= increment;
    }
```

Maybe kind of a hack, it just says at least move up or down one full line minimum.

**Comment by dterrahe** [comment]
Try moving the part that @johnny-bit added in #7108
```
    int delta_y=0;

    dt_gui_get_scroll_unit_deltas(event, NULL, &delta_y);
```
outside the `if` and use `delta_y` instead of `event->delta_y`?

**Comment by TurboGit** [comment]
@dterrahe : No luck, delta_y is 0 when I do slow scroll.

**Comment by dterrahe** [comment]
But it should add up to when you've accumulated a full unit and then scroll one step? Doesn't this happen for you elsewhere where dt_gui_get_scroll_unit_deltas is used?

**Comment by TurboGit** [comment]
For me the the adjustment value -= value % increment; is wrong when going down. This really works well for a single direction or when the delta is always >= 1.0 but for mouse scroll with high sensibility it doesn't work. Or I'm really tired tonight :)

See #11122

**Comment by TurboGit** [comment]
@dterrahe : I suppose all this come down to the values `increment` and `delta_y`. What values have you for both of those variables?

**Comment by dterrahe** [comment]
increment will depend on font size, but really only serves to keep the list aligned on word or item boundaries, so should be somewhat irrelevant to this issue.

**Comment by dterrahe** [comment]
If you feed `event` through `dt_gui_get_scroll_unit_deltas` you should be getting 0, 0, 0, 1, 0, 0, etc so we shouldn't have to deal with "small < 1.0" anymore? I'm confused...

**Comment by TurboGit** [comment]
@dterrahe: New version proposed, I have found the issue. Now I have the scroll up/down behaving exactly the same way with the same speed and no jerky jumping around. Hope this one is better, but I do feel so.
